src/care/algorithm_decl.h

```diff
@@ -21,10 +21,20 @@
 #include "LLNL_GlobalID.h"
 #endif // CARE_HAVE_LLNL_GLOBALID
 
-#define CARE_MAX(a,b) a > b ? a : b
-#define CARE_MIN(a,b) a < b ? a : b
-
 namespace care {
+
+template <typename T>
+CARE_HOST_DEVICE CARE_INLINE T max(const T a, const T b)
```

Comment:
Can you use references to match the signature of std::max?

```
template<class T> 
const T& max(const T& a, const T& b)
```

Reply:
I guess that could prohibit using literals, though, so maybe not.

Reply:
whoa I had no idea max returned a reference. But makes sense when you start to think about it from a performance / avoiding copy construction perspective.

Reply:
https://en.cppreference.com/w/cpp/algorithm/max

They have examples with literals - I'm not sure if those work with the templated &  signature and the inlining makes it all happy or if it ends up going through the initializer list signature when you use literals.

Reply:
I think we should try matching the standard signature.

Reply:
https://en.cppreference.com/w/cpp/language/reference_initialization
I think this explains why this works